roxctl/generate/netpol/cta.go

```diff
@@ -5,24 +5,54 @@ import (
 	"path/filepath"
 	"strings"
 
-	"github.com/np-guard/cluster-topology-analyzer/pkg/controller"
+	npguard "github.com/np-guard/cluster-topology-analyzer/pkg/controller"
 	"github.com/pkg/errors"
 	"github.com/stackrox/rox/pkg/protoconv/networkpolicy"
 	v1 "k8s.io/api/networking/v1"
 )
 
+var errNPGErrorsIndicator = errors.New("there were errors during execution")
+
 func (cmd *generateNetpolCommand) generateNetpol() error {
-	recommendedNetpols, err := controller.PoliciesFromFolderPath(cmd.folderPath)
+	opts := []npguard.PoliciesSynthesizerOption{
+		npguard.WithLogger(newNpgLogger(cmd.env.Logger())),
+	}
+	if cmd.stopOnFirstError {
+		opts = append(opts, npguard.WithStopOnError())
+	}
+	npSynth := npguard.NewPoliciesSynthesizer(opts...)
+	recommendedNetpols, err := npSynth.PoliciesFromFolderPath(cmd.folderPath)
 	if err != nil {
-		return errors.Wrap(err, "error synthesizing policies from folder")
+		return errors.Wrap(err, "error generating network policies")
+	}
+	if err := cmd.ouputNetpols(recommendedNetpols); err != nil {
+		return err
+	}
+	var roxerr error = nil
+	for _, e := range npSynth.Errors() {
+		if e.IsSevere() {
+			cmd.env.Logger().ErrfLn("%s %s", e.Error(), e.Location())
+			roxerr = errNPGErrorsIndicator
+		} else {
+			cmd.env.Logger().WarnfLn("%s %s", e.Error(), e.Location())
+			if cmd.treatWarningsAsErrors {
+				roxerr = errNPGErrorsIndicator
+			}
+		}
 	}
+	return roxerr
+}
+
+func (cmd *generateNetpolCommand) ouputNetpols(recommendedNetpols []*v1.NetworkPolicy) error {
 	if _, err := os.Stat(cmd.outputFolderPath); err == nil {
 		if err := os.RemoveAll(cmd.outputFolderPath); err != nil {
```

Comment:
Should we ask user about removing output first? What if by mistake user pass a file or `/` instead of `./`

Reply:
This happens only when the user asks for it explicitly by providing `--remove` flag.
See https://github.com/stackrox/stackrox/blob/edfb373fe06d368440609f02b30e24003870cf9a/roxctl/generate/netpol/netpol.go#L82-L89